### PR TITLE
MCOL-5623 This patch changes the logic applied to `save` function for EM

### PR DESCRIPTION
**Pull request by denis0x0D**

### CODE REVIEW

File: versioning/BRM/extentmap.cpp [hunk 1]
```diff
@@ -1802,26 +1802,28 @@ void ExtentMap::save(const string& filename)
 
 #endif
 
-  grabEMEntryTable(READ);
-  grabEMIndex(READ);
+  // MCOL-5623 Grag the `WRITE` lock, we cannot let BRM sctructres to be modified (new Extent), while saving
+  // them to the file.
+  grabEMEntryTable(WRITE);
```
Comment (drrtuy) [inline versioning/BRM/extentmap.cpp]: READ lock taken assumes there will be no writes. 
Taking a WRITE lock you disallows other threads to read EM or EMIndex.

**Comment by denis0x0D** [comment]
When SM is enabled one `write` takes about ~2 milliseonds, so if em has about 150k elements, it takes about 300 seconds to `save` function instead ~4 milliseconds.

File: versioning/BRM/extentmap.cpp [hunk 2]
```diff
@@ -1802,26 +1802,28 @@ void ExtentMap::save(const string& filename)
 
 #endif
 
-  grabEMEntryTable(READ);
-  grabEMIndex(READ);
+  // MCOL-5623 Grag the `WRITE` lock, we cannot let BRM sctructres to be modified (new Extent), while saving
```
Comment (drrtuy) [inline versioning/BRM/extentmap.cpp]: Type in Grag I suppose.

**Review by drrtuy — CHANGES_REQUESTED** [review]
Take a look at the comments plz.

File: versioning/BRM/extentmap.cpp (hunk 1 above)
Comment (denis0x0D) [reply versioning/BRM/extentmap.cpp]: I see, thanks for showing this, fixed

**Review by drrtuy — APPROVED** [review]
Untested ACK

**Pull request merged**